msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,333 @@
+* Messaging extensions must respond to user commands within five seconds.
+* Notification must load within five seconds of the user action.
+
+### 3.4 Platform interoperability
+
+* Apps must be fully responsive to a variety of screen sizes.
+* App must not include any direct references to in-app purchases, trial offers, offers for paid versions, or links to any online stores where users can purchase or acquire other content, apps, or add-ins from your Teams app on mobile OS (Android, iOS).
+* The iOS or Android version of the app must not show any UI or language or link to any other apps, add-ins, or websites that ask the user to pay.
+* The associated privacy policy and terms of use pages must also be free of any commerce UI or store links.
+* If your app contains tabs, they must follow the mobile design guidelines.
+* App should function on the following platforms:
+    * iOS 9.0 and later
+    * Android 4.4 and later
+    * Windows Phone 10.0.10586 and later
+
+## 4.0 App package and store listing
+
+App packages must be correctly formatted and include all required information and components.
+
+### 4.1 App manifest
+
+* The app manifest must conform to the [latest manifest schema](~/resources/schema/manifest-schema.md).
+* If your app includes a bot or messaging extension, the manifest must be consistent with Bot Framework metadata, including bot name, logo, privacy policy link, and terms of service link.
+* If your app uses Azure Active Directory (Azure AD) for authentication, include the Azure AD ID in the manifest. For more information, see [manifest schema](~/resources/schema/manifest-schema.md#webapplicationinfo).
+
+### 4.2 App icons
+
+* Your app package must include two PNG versions of your app icon: a color icon and an outline icon.
+* Correctly sized and formatted icons must be specified in your app package. The icons must match what's submitted with the store listing metadata. For more information, see the Teams app [icon guidelines](~/concepts/build-and-test/apps-package.md#app-icons).
+
+### 4.3 App description
+
+The app descriptions in your manifest and store must be the same and follow these guidelines:
+
+* Use "Microsoft Teams" on first reference. Subsequent references can be shortened to ”Teams”.
+* Do not abbreviate "Microsoft" (such as "MS" or "MSFT").
+* Avoid unnecessary capitalization and language errors.
+* Follow the [short description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#short-description).
+* Follow the [long description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#long-description).
+* Clearly describe any limitations, conditions, or exceptions to the functionality, features, and deliverables described in the listing and related materials before the customer acquires your offer. The capabilities you declare must relate to the core functions and description of your offer.
+
```

Comment:
@heath-hamilton Add -
4.5 Privacy policy
• The privacy policy can be specific to your app and/or an overall policy for all of your services.
• If you use a generic privacy policy, it must reference "services", "applications", and "platforms" to include your

4.6 Teams app as well as your website.
• It must include how you handle user data storage, user data retention, deletion, and security controls.
• It must include your contact information.
• It should not contain broken links, beta URLs, or staging URLs.

4.7 Terms of use
Your terms of use statement should be specific and applicable to your app and/or add-in offering.
Support URLs
Your support URLs should not require authentication or login credential to contact you for any issues with your app.